### DISCUSSION TITLE
Is there any possible way to convert a value to std::string? or, to expand, convert value to any types whatever we want, if is accessible?

### DISCUSSION THREAD

**Discussion by laper32** [nlohmann/json#2531] [Q&A] [answered]
For example, here is a json here:
```
{
    "pi": 3.14,
    "key1": "str"
}
```

in some terms, for example, to construct string, we need to convert all of values into string, for example: SQL.
then, in theorically, we can assume that there is a SELECT script what we need to construct here:
```sql
SELECT (whatever fields, it doesn't affect anything here.) FROM (whatever tables, it does not affect anything here.) WHERE `pi` = 3.14 AND `key1` = "str"
```

In my memory, here, `pi`=3.14 can equal to "3.14", ie: `pi`="3.14", then we can rewrite the script to:
```sql
SELECT (whatever fields, it doesn't affect anything here.) FROM (whatever tables, it does not affect anything here.) WHERE `pi` = "
\"3.14\" AND `key1` = \"str\"
```
here is the problem: is there any possible way to convert float/int/boolean/etc(if acceptable) to std::string, or, reversely, convert std::string to float/int/boolean/etc(if acceptable)?

Note: in this term, we dont know what the input typename is. That is, the json string can be arbitrary, ie: if we modify something here:
```json
{
    "boolean": 1,
    "key1": "str",
    "float": 1.15
}
```
it should work as previous one(we are not considering array/object here).
I've read previous issue/discussion already, but sadly that unable to solve my problem.

thanks for answering.

**Comment by nlohmann** [comment, marked as answer]
The library does not provide to convert arbitrary types to string - this must be done in the client application. However, maybe you can use `dump()` to create a JSON text out of your values. So in your example, assuming the JSON value is in variable `j`:

```cpp
std::string s1 = j["boolean"].dump();
std::string s2 = j["float"].dump();
```

This should yield something like:

```
s1 == "1";
s2 == "1.15";
```
